bench/locli/src-quick/Cardano/Analysis/Reducer.hs

```diff
@@ -0,0 +1,72 @@
+
+module Cardano.Analysis.Reducer
+       (module Cardano.Analysis.Reducer)
+       where
+
+import           Cardano.Api (SlotNo)
+
+import           Cardano.Analysis.Reducer.Util
+import           Cardano.Unlog.LogObject (LOBody (..), LogObject (..))
+import           Cardano.Util
+
+import           Prelude hiding (log, seq)
+
+import           Data.Reducer
+import           Data.Word
+
+
+data TxsInMempool = TxsInMempoolPerSlot
+
+instance Reducer TxsInMempool where
+  type instance Elem   TxsInMempool = BySlot LogObject
+  type instance Accum  TxsInMempool = ([(SlotNo, Word64)], Word64)
+  type instance Result TxsInMempool = [(SlotNo, Word64)]
+
+  initialOf _ = ([], 0)
+
+  -- every slot (x-axis) contains one value on the y-axis (ideal for plotting a curve)
+  reducerOf _ (acc, currentTxns) (BySlot (slotNo, objs)) =
+    case safeLast objs of
+      [ LogObject{loBody = LOMempoolTxs txns} ]
+        -> ((slotNo, txns) : acc, txns)
+      _ -> ((slotNo, currentTxns) : acc, currentTxns)
+
+  resultOf _ = reverse . fst
+
+
+data ResourceMeasure = ResourceMeasurePerSlot
+
+instance Reducer ResourceMeasure where
+  type instance Elem   ResourceMeasure = BySlot Word64
+  type instance Accum  ResourceMeasure = [(SlotNo, Word64)]
+  type instance Result ResourceMeasure = [(SlotNo, Word64)]
+
+  initialOf _ = []
+
+  reducerOf _ acc (BySlot (slotNo, objs)) =
+    case safeLast objs of
+      [measurement] -> (slotNo, measurement) : acc
+      _             -> acc
+
+  resultOf _ = reverse
+
+
+data Silence = Silence {threshold :: NominalDiffTime, startTime :: UTCTime}
```

Comment:
Add strictness annotations to the fields, and use `seq` in the `Reducer` instance

Reply:
Thanks, added. About `seq`: currently Haskell lists are used still to ease development. Once the type is changed to something more fitting, an eval strategy will be chosen that goes best with it.